verification/curator-service/api/src/controllers/auth.ts

```diff
@@ -158,59 +208,148 @@ export class AuthController {
             },
         );
 
+        /**
+         * Generate reset password link
+         */
         this.router.post(
-            '/signup',
-            async (req: Request, res: Response): Promise<void> => {
-                let user = await User.findOne({
-                    email: req.body.email,
-                }).exec();
+            '/request-password-reset',
+            async (req: Request, res: Response): Promise<Response<any>> => {
+                const email = req.body.email as string;
 
-                // Update newsletter preferences
-                if (
-                    user &&
-                    user.newsletterAccepted === false &&
-                    req.body.newsletter
-                ) {
-                    user = await User.findOneAndUpdate(
-                        { email: req.body.email },
-                        { newsletterAccepted: req.body.newsletter },
+                try {
+                    // Check if user with this email address exists
+                    const user = await User.findOne({ email });
+                    if (!user) {
+                        return res.sendStatus(200);
+                    }
+
+                    // Check if user is a Gmail user and send appropriate email message in that case
+                    if (user.googleID !== '42') {
+                        await this.emailClient.send(
+                            [email],
+                            'Password reset requested',
+                            `We were asked to reset your password on Global.health, but you are registered with your Google account. 
+                            If you requested the password reset, then please try again, this time using the "Sign in with Google" button on the global.health portal. 
+                            Otherwise, no further action is needed.`,
+                        );
+
+                        return res.sendStatus(200);
+                    }
+
+                    // Check if there is a token in DB already for this user
+                    const token = await Token.findOne({ userId: user._id });
+                    if (token) {
+                        await token.deleteOne();
+                    }
+
+                    const resetToken = crypto.randomBytes(32).toString('hex');
+                    const hash = await bcrypt.hash(resetToken, 10);
+
+                    // Add new reset token to DB
+                    await new Token({
+                        userId: user._id,
+                        token: hash,
+                        createdAt: Date.now(),
+                    }).save();
+
+                    let url = '';
+                    switch (env) {
+                        case 'local':
+                            url = 'http://localhost:3002';
+                            break;
+                        case 'dev':
+                            url = 'https://dev-curator.ghdsi.org';
```

Comment:
should be https://dev-data.covid-19.global.health

Reply:
Thanks, already updated